packages/nimble-components/src/text-area/index.ts

```diff
@@ -26,6 +27,67 @@ export class TextArea extends FoundationTextArea {
      */
     @attr
     public appearance: TextAreaAppearance = TextAreaAppearance.outline;
+
+    /**
+     * A message explaining why the value is invalid.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-text
+     */
+    @attr({ attribute: 'error-text' })
+    public errorText?: string;
+
+    /**
+     * Whether to display the error state.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-visible
+     */
+    @attr({ attribute: 'error-visible', mode: 'boolean' })
+    public errorVisible = false;
+
+    /**
+     * The width of the vertical scrollbar, if displayed.
+     * @internal
+     */
+    @observable
+    public scrollbarWidth = 0;
+
+    private resizeObserver?: ResizeObserver;
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.resizeObserver = new ResizeObserver(this.onResize);
+        this.resizeObserver.observe(this);
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.disconnect();
+    }
+
+    /**
+     * @internal
+     */
+    public onTextAreaInput(): void {
+        this.handleTextInput();
+        this.updateScrollbarWidth();
+    }
+
+    private readonly onResize = (): void => {
+        this.updateScrollbarWidth();
+    };
+
+    private updateScrollbarWidth(): void {
+        this.scrollbarWidth = this.control.offsetWidth - this.control.clientWidth;
+    }
```

Comment:
As currently implemented the assumption is that only the control size changing or the user updating the value will result in the layout of the contents shifting (resulting in scrollbars being added or removed).

But I think potentially many property changes can result in that, i.e. changing the placeholder when it is visible, changing the value programmatically, changing the errorVisible state.

Here is an example of the errorVisible state turning on and failing to render correctly because scrollbars are added when errorVisible is set to true:

![erroricon](https://user-images.githubusercontent.com/1588923/228616494-2ba924c7-d1d7-45a3-a454-675b1bf924c2.gif)

Curious on opinions @jattasNI @m-akinc 

1. Do we want to play whack-a-mole on this? We can add a errorVisibleChanged lifecycle hook for the case I show. I'm concerned about performance if doing it on something that can change rapidly like the value as measuring `this.control.offsetWidth - this.control.clientWidth` will perform a page layout outside a ResizeObserver.
2. Do we accept that we are only responding to specific properties and that others are a known issue. Do we document that in a comment? Do we document that in storybook somehow? Do we cite a workaround like if you change properties make sure to toggle errorVisible on and off so the icon is positioned correctly.

Reply:
My personal preference is 4.  I think as the user is typing if the app is doing validation as the user is typing the error-icon may appear and cause the contents that they are editing to shift on the screen. That feels like a poor user experience that as they type the error icon can pop in and out causing the contents to reflow.

Reply:
The UX argument about the text shifting as it goes in and out of the error state is compelling. If we were okay with that, though, I could try the approach you suggested before about setting the icon as the textarea's background. I think that would avoid the whack-a-mole issues. We could also avoid the UX issue if we always reserve the space for the error icon. Maybe that wouldn't be so bad.

I asked Brandon about this, and here's his response:
![image](https://user-images.githubusercontent.com/7282195/228630180-eb355811-b514-4365-bdfd-dfe14d86f08c.png)
![image](https://user-images.githubusercontent.com/7282195/228630575-516a8edb-2c05-4ef5-a005-30fbed11de1b.png)

I personally really like the consistency of having the error icon in the upper right. I'm leaning towards keeping it (assuming the bg image approach works), but I'm not sure whether I'd rather always reserve space for it or not.

Reply:
The compromise we came up with was to move the icon as far to the right as possible, and always reserve enough space at the right side of the text box to show the icon without overlapping any text. Reserving the space allows the error state to flicker in and out without affecting the layout of the text.